source/content/guides/secure-development/05-two-factor-authentication.md

```diff
@@ -0,0 +1,255 @@
+
+1. Configure the **Attribute Mapping** in the SSO/SAML Settings of the WordPress Admin with the values shown below. Values are case-sensitive.
+
+  ![TFA OneLogin Attributes](../../../images/tfa-wp-onelogin-attribute.png)
+
+1. Configure the **Customize Actions and Links** in the SSO/SAML Settings of the WordPress Admin to **Prevent use of ?normal**. This requires OneLogin as the authentication solution.
+
+  ![TFA OneLogin Custom Actions](../../../images/tfa-onelogin-custom-actions.png)
+
+1. Use the OneLogin dashboard to log in to your WordPress site.
+
+  ![TFA OneLogin WP Login](../../../images/tfa-onelogin-wp-login.png)
+
+  <Alert title="Note" type="info">
+
+  You can also refer to OneLogin's documentation, [Configure SAML for WordPress](https://onelogin.service-now.com/support/?id=kb_article&sys_id=590b1d83db109700d5505eea4b96197f), for further troubleshooting. You will need a OneLogin admin account to access their knowledge base.
+
+  </Alert>
+
+</Tab>
+
+<Tab title="Drupal" id="drupal-org">
+
+There are many different organization-wide [Drupal modules for single sign on](https://groups.drupal.org/node/182004) that can also provide TFA capabilities. One of the service options we use internally at Pantheon is OneLogin, which has the [OneLogin](https://www.drupal.org/project/onelogin) module.
+
+#### OneLogin Instructions
+
+1. Sign up and create a [OneLogin account](https://www.onelogin.com/) for your organization.
+
+1. Install the Drupal SAML 2.0 app connector as part of the OneLogin dashboard. This will need to be done for each Drupal site that is being managed by OneLogin.
+
+1. Edit the OneLogin Drupal app connector to provide the appropriate default values for the Configuration section. Other sections should already be set up correctly.
+  ![TFA OneLogin Config](../../../images/tfa-drupal-onelogin-config.png)
+
+1. **(Optional)** Configure the **Authentication Factors** found under Settings for a list of authentication factors you can enable for your different users.
+
+  ![TFA OneLogin Methods](../../../images/tfa-onelogin-tfa-methods.png)
+
+1. Create user accounts in the Users Administration area in OneLogin.
+
+1. Click **New User** and verify that the “Username” and "Email" fields in OneLogin match their Drupal username and email.
+
+![TFA OneLogin New User](../../../images/tfa-onelogin-new-user.png)
+
+#### Drupal Instructions
+
+1. Install and enable the GitHub version of the [OneLogin SAML](https://github.com/onelogin/drupal-saml) module on your Drupal site. This module is eventually intended to live on Drupal.org as the [2.x branch of the OneLogin project](https://www.drupal.org/project/onelogin).
+
+1. Set the `$_SERVER['SERVER_PORT']` value in `settings.php` according to [these instructions](/server_name-and-server_port). This change is necessary to have SAML use the appropriate ports.
+
+1. Configure the OneLogin SAML module `admin/config/onelogin_saml` to match the screenshot below. Values are case-sensitive.
```

Comment:
Same as above: The values need to be included as text, for accessibility purposes.  I would have done so myself, but wasn't sure if they required any placeholder values.  Also, once the values are included as text, you may not need the image anymore.